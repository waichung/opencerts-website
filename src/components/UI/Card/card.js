import PropTypes from "prop-types";
import css from "./card.scss";

const Card = props => (
  <div
    id={props.info.id}
    className={`col-lg-4 col-md-6 col-sm-12 ${css["mb-3"]} ${css["m-pd-0"]}`}
    style={{ paddingBottom: "36px", wordWrap: "break-word" }}
  >
    <a
      href={props.info.website}
      target="_blank"
      rel="noopener noreferrer nofollow"
      style={{ color: "#000", textDecoration: "none" }}
    >
      <div className={css["partner-block"]}>
        <img
          className={`${css.logo}`}
          src={props.info.logo}
          id={css[props.info.key]}
        />
        <h4 className={css["partner-name"]} style={{ fontWeight: "bold" }}>
          {props.info.name}
        </h4>
        {props.info.address ? (
          <div>
            Certificate Store:{" "}
            <a href={`https://etherscan.io/address/${props.info.address}`}>
              {props.info.address}
            </a>
          </div>
        ) : (
          ""
        )}
        {props.info.address && props.info.website ? (
          <div>
            Website:{" "}
            <a
              href={props.info.website}
              target="_blank"
              rel="noopener noreferrer"
            >
              {props.info.website}
            </a>
          </div>
        ) : (
          ""
        )}
        {props.info.email ? (
          <div>
            Email: <a href={`mailto:${props.info.email}`}>{props.info.email}</a>
          </div>
        ) : (
          ""
        )}
        {props.info.phone ? (
          <div>
            Phone: <a href={`tel:${props.info.phone}`}>{props.info.phone}</a>
          </div>
        ) : (
          ""
        )}
        {props.info.description ? <div>{props.info.description}</div> : ""}
      </div>
    </a>
  </div>
);

export default Card;

Card.propTypes = {
  info: PropTypes.object
};
