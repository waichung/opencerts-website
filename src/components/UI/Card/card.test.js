import React from "react";
import renderer from "react-test-renderer";
import Card from "./card";

describe("default card", () => {
  it("matches snapshot", () => {
    const fakeData = {
      id: "Test Id",
      name: "Test Name",
      description: "Test Description",
      website: "Test Website",
      email: "Test Email",
      phone: "Test Phone",
      key: "Test Key",
      logo: "Test Logo"
    };
    const component = renderer.create(<Card info={fakeData} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
