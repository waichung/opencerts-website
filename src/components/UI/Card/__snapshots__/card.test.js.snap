// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default card matches snapshot 1`] = `
<div
  className="col-lg-4 col-md-6 col-sm-12 undefined undefined"
  id="Test Id"
  style={
    Object {
      "paddingBottom": "36px",
      "wordWrap": "break-word",
    }
  }
>
  <a
    href="Test Website"
    rel="noopener noreferrer nofollow"
    style={
      Object {
        "color": "#000",
        "textDecoration": "none",
      }
    }
    target="_blank"
  >
    <div>
      <img
        className="undefined"
        src="Test Logo"
      />
      <h4
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Test Name
      </h4>
      
      
      <div>
        Email: 
        <a
          href="mailto:Test Email"
        >
          Test Email
        </a>
      </div>
      <div>
        Phone: 
        <a
          href="tel:Test Phone"
        >
          Test Phone
        </a>
      </div>
      <div>
        Test Description
      </div>
    </div>
  </a>
</div>
`;
